--- BF2556X-1T_BSP_9.0.0-master.orig/platforms/apsn/src/bf_pltfm_led/bf_pltfm_mav_led.c	2019-11-19 16:00:22.000000000 +0100
+++ BF2556X-1T_BSP_9.0.0-master/platforms/apsn/src/bf_pltfm_led/bf_pltfm_mav_led.c	2020-11-20 10:20:39.053542297 +0100
@@ -1018,7 +1018,7 @@
                             /* get speed info */
                             port_hdl.conn_id = bf_pltfm_port_runtime_info[i].conn_id;
                             port_hdl.chnl_id = bf_pltfm_port_runtime_info[i].chnl_id;
-                            bf_pm_port_front_panel_port_to_dev_port_get(0, &port_hdl, &dev_port);
+                            if (bf_pm_port_front_panel_port_to_dev_port_get(0, &port_hdl, &dev_port) == BF_SUCCESS) {
                             if(strstr(cpss_buf, "1G"))
                             {
                                 bf_pltfm_port_runtime_info[i].port_speed = BF_PLTFM_PORT_SPEED_1G;
@@ -1046,6 +1046,7 @@
                             bf_pltfm_port_runtime_info[i].link_status_change = (bf_pltfm_port_runtime_info[i].link_status==0)?1:0;
                             bf_pltfm_port_runtime_info[i].link_status = 1;
                             get_port_speed_flag = 1;
+                            }
                         }
                         
                         break;
