diff -ruN BF2556X-1T_BSP_9.0.0-master.orig/bf-platforms-9.0.0/platforms/delta-bf/include/bf_mav_led.h BF2556X-1T_BSP_9.0.0-master/bf-platforms-9.0.0/platforms/delta-bf/include/bf_mav_led.h
--- BF2556X-1T_BSP_9.0.0-master.orig/bf-platforms-9.0.0/platforms/delta-bf/include/bf_mav_led.h	2019-11-19 16:00:22.000000000 +0100
+++ BF2556X-1T_BSP_9.0.0-master/bf-platforms-9.0.0/platforms/delta-bf/include/bf_mav_led.h	2020-11-20 12:51:31.701363265 +0100
@@ -40,6 +40,71 @@
 #define BF_MAV_PORT_LED_BLUE 0x1
 #define BF_MAV_PORT_LED_BLINK 0x8
 
+#include <bf_pltfm_types/bf_pltfm_types.h>
+#include <bf_led/bf_led.h>
+
+#define BF_PLTFM_PORT_NUM_MAX                               (80)
+#define BF_PLTFM_PIPE_PORT_NUM                              (16)
+
+typedef enum bf_pltfm_led_color_t
+{
+    BF_PLTFM_LED_COLOR_BLACK=0,
+    BF_PLTFM_LED_COLOR_BLUE,
+    BF_PLTFM_LED_COLOR_GREEN,
+    BF_PLTFM_LED_COLOR_CYAN,
+    BF_PLTFM_LED_COLOR_RED,
+    BF_PLTFM_LED_COLOR_MAGENTA,
+    BF_PLTFM_LED_COLOR_YELLOW,
+    BF_PLTFM_LED_COLOR_WHITE,
+    BF_PLTFM_LED_COLOR_MAX
+}BF_PLTFM_LED_COLOR_T;
+
+typedef enum bf_pltfm_led_mode_t
+{
+    BF_PLTFM_LED_MODE_OFF=0,
+    BF_PLTFM_LED_MODE_ON,
+    BF_PLTFM_LED_MODE_FAST_BLINK,
+    BF_PLTFM_LED_MODE_SLOW_BLINK,
+    BF_PLTFM_LED_MODE_MAX
+}BF_PLTFM_LED_MODE_T;
+
+typedef enum bf_pltfm_port_speed_t
+{
+    BF_PLTFM_PORT_SPEED_NONE=-1,
+    BF_PLTFM_PORT_SPEED_1G=0,
+    BF_PLTFM_PORT_SPEED_2G_PLUS_500M,
+    BF_PLTFM_PORT_SPEED_5G,
+    BF_PLTFM_PORT_SPEED_10G,
+    BF_PLTFM_PORT_SPEED_25G,
+    BF_PLTFM_PORT_SPEED_40G,
+    BF_PLTFM_PORT_SPEED_50G,
+    BF_PLTFM_PORT_SPEED_100G,
+    BF_PLTFM_PORT_SPEED_MAX
+}BF_PLTFM_PORT_SPEED_T;
+
+typedef struct bf_pltfm_port_speed_led_t
+{
+    BF_PLTFM_PORT_SPEED_T port_speed;
+    BF_PLTFM_LED_COLOR_T led_color;
+    BF_PLTFM_LED_MODE_T blink_mode;
+}BF_PLTFM_PORT_SPEED_LED_T;
+
+typedef struct bf_pltfm_port_runtime_info_t
+{
+    uint32_t conn_id;
+    uint32_t chnl_id;
+    uint32_t pres_status;
+    uint32_t link_status;
+    uint32_t link_status_change;
+    BF_PLTFM_PORT_SPEED_T port_speed;
+    uint64_t last_tx_counter;
+    uint64_t last_rx_counter;
+    uint32_t led_blink;
+    uint8_t last_led_bits;
+}BF_PLTFM_PORT_RUNTIME_INFO_T;
+
+extern void bf_pltfm_port_runtime_info_callback(const BF_PLTFM_PORT_RUNTIME_INFO_T *info_array);
+
 /* display port led using tofino stateout buffer */
 int bf_pltfm_port_led_by_tofino_set(int chip_id,
                                     bf_pltfm_port_info_t *port_info,
diff -ruN BF2556X-1T_BSP_9.0.0-master.orig/bf-platforms-9.0.0/platforms/delta-bf/src/bf_pltfm_led/bf_pltfm_mav_led.c BF2556X-1T_BSP_9.0.0-master/bf-platforms-9.0.0/platforms/delta-bf/src/bf_pltfm_led/bf_pltfm_mav_led.c
--- BF2556X-1T_BSP_9.0.0-master.orig/bf-platforms-9.0.0/platforms/delta-bf/src/bf_pltfm_led/bf_pltfm_mav_led.c	2020-11-20 10:22:33.705540029 +0100
+++ BF2556X-1T_BSP_9.0.0-master/bf-platforms-9.0.0/platforms/delta-bf/src/bf_pltfm_led/bf_pltfm_mav_led.c	2020-11-20 10:46:08.989512040 +0100
@@ -111,8 +111,6 @@
 #define BF_LED_STATEOUT_BUF_SIZE 128
 
 #define BF_PLTFM_PORT_LED_I2C_ADDR                          (0x32)
-#define BF_PLTFM_PORT_NUM_MAX                               (80)
-#define BF_PLTFM_PIPE_PORT_NUM                              (16)
 #define BF_PLTFM_PORT_COUNTER_UPDATE_INTERVAL_SEC           (3)
 #define BF_PLTFM_I2C_PORT_SCAN_RETRY                        (3)
 #define bF_PLTFM_IRQ_REG_FILE_PATH                          "/sys/kernel/debug/reg_pid_for_irq19"
@@ -155,62 +153,6 @@
         pres_value = (!(port_1_32_pres & (0x1 << (port_index2-36))))?1:0; \
     }
     
-typedef enum bf_pltfm_led_color_t
-{
-    BF_PLTFM_LED_COLOR_BLACK=0,
-    BF_PLTFM_LED_COLOR_BLUE,
-    BF_PLTFM_LED_COLOR_GREEN,
-    BF_PLTFM_LED_COLOR_CYAN,
-    BF_PLTFM_LED_COLOR_RED,
-    BF_PLTFM_LED_COLOR_MAGENTA,
-    BF_PLTFM_LED_COLOR_YELLOW,
-    BF_PLTFM_LED_COLOR_WHITE,
-    BF_PLTFM_LED_COLOR_MAX
-}BF_PLTFM_LED_COLOR_T;
-
-typedef enum bf_pltfm_led_mode_t
-{
-    BF_PLTFM_LED_MODE_OFF=0,
-    BF_PLTFM_LED_MODE_ON,
-    BF_PLTFM_LED_MODE_FAST_BLINK,
-    BF_PLTFM_LED_MODE_SLOW_BLINK,
-    BF_PLTFM_LED_MODE_MAX
-}BF_PLTFM_LED_MODE_T;
-
-typedef enum bf_pltfm_port_speed_t
-{
-    BF_PLTFM_PORT_SPEED_NONE=-1,
-    BF_PLTFM_PORT_SPEED_1G=0,
-    BF_PLTFM_PORT_SPEED_2G_PLUS_500M,
-    BF_PLTFM_PORT_SPEED_5G,
-    BF_PLTFM_PORT_SPEED_10G,
-    BF_PLTFM_PORT_SPEED_25G,
-    BF_PLTFM_PORT_SPEED_40G,
-    BF_PLTFM_PORT_SPEED_50G,
-    BF_PLTFM_PORT_SPEED_100G,
-    BF_PLTFM_PORT_SPEED_MAX
-}BF_PLTFM_PORT_SPEED_T;
-
-typedef struct bf_pltfm_port_speed_led_t
-{
-    BF_PLTFM_PORT_SPEED_T port_speed;
-    BF_PLTFM_LED_COLOR_T led_color;
-    BF_PLTFM_LED_MODE_T blink_mode;
-}BF_PLTFM_PORT_SPEED_LED_T;
-
-typedef struct bf_pltfm_port_runtime_info_t
-{
-    uint32_t conn_id;
-    uint32_t chnl_id;
-    uint32_t pres_status;
-    uint32_t link_status;
-    uint32_t link_status_change;
-    BF_PLTFM_PORT_SPEED_T port_speed;
-    uint64_t last_tx_counter;
-    uint64_t last_rx_counter;
-    uint32_t led_blink;
-    uint8_t last_led_bits;
-}BF_PLTFM_PORT_RUNTIME_INFO_T;
 
 enum 
 {
@@ -1207,6 +1149,8 @@
 
 }
 
+
+
 static void *bf_pltfm_port_update_runtime_info()
 {
     while(bf_pltfm_port_runtime_thread_run==1)
@@ -1228,6 +1172,7 @@
             BF_PLTFM_PORT_DEBUG("BF_PLTFM_PORT_STATUS_ANAYLYZIED");
             bf_pltfm_port_status_analyze = BF_PLTFM_PORT_STATUS_ANAYLYZIED;
         }
+        bf_pltfm_port_runtime_info_callback(bf_pltfm_port_runtime_info);
         bf_sys_usleep(100000);
     }
     
