From 9eec589b141a049f7afff5bab51d46d167a74d7d Mon Sep 17 00:00:00 2001
From: Mattias Waldo <mattias.waldo@saabgroup.com>
Date: Fri, 27 Nov 2020 12:32:33 +0100
Subject: [PATCH] Relax file existence tests to allow cross-compile.

---
 configure.ac       | 5 +++--
 proto/configure.ac | 5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 4b09a74..9fed725 100644
--- a/configure.ac
+++ b/configure.ac
@@ -65,8 +65,9 @@ AM_CONDITIONAL([WITH_GTEST], [test "$with_proto" = yes])
 AM_COND_IF([WITH_GTEST], [
     gtest_root=$ac_abs_confdir/third_party/googletest
     gtest_check_f=$gtest_root/googletest/src/gtest.cc
-    AC_CHECK_FILE([$gtest_check_f], [],
-                  [AC_MSG_ERROR([Did you run 'git submodule update --init'?])])
+    if ! test -f $gtest_check_f ; then
+       AC_MSG_ERROR([Did you run 'git submodule update --init'?])
+    fi
 ])
 
 AC_ARG_WITH([internal-rpc],
diff --git a/proto/configure.ac b/proto/configure.ac
index aaa2707..060f5b2 100644
--- a/proto/configure.ac
+++ b/proto/configure.ac
@@ -83,8 +83,9 @@ AM_COND_IF([WITH_SYSREPO], [
 ])
 
 p4runtime_check_f=$ac_abs_confdir/p4/v1/p4runtime.proto
-AC_CHECK_FILE([$p4runtime_check_f], [],
-              [AC_MSG_ERROR([Cannot find p4runtime; did you run 'git submodule update --init'?])])
+if ! test -f $p4runtime_check_f ; then
+   AC_MSG_ERROR([Cannot find p4runtime; did you run 'git submodule update --init'?])
+fi
 
 AC_ARG_ENABLE([Werror],
     AS_HELP_STRING([--enable-Werror], [Make all compiler warnings fatal]),
-- 
2.25.1

