--- bf-drivers/src/pi/pi_tables_imp.c	2020-04-30 19:39:38.000000000 +0200
+++ bf-drivers/src/pi/pi_tables_imp.c	2020-12-07 14:28:51.625620236 +0100
@@ -1279,6 +1279,12 @@
         "%s: failed to retrieve match entry from handle, "
         "dropping idle time notification",
         __func__);
+
+    status = pipe_mgr_client_cleanup(sess);
+    if (status != PIPE_SUCCESS) {
+      LOG_CRIT("%s: failed to close pipe_mgr session", __func__);
+    }
+
     pi_state_unlock(pi_dev_id);
     return;
   }
